source/plugin/blocks/tileentities.rst

```diff
@@ -66,21 +67,20 @@ line, the second attempts to set it and returns the boolean value indicating its
     public Optional<Text> getFirstLine(TileEntity entity) {
         Optional<SignData> data = entity.getOrCreate(SignData.class);
         if (data.isPresent()) {
-            return Optional.of(data.get().getLine(0));
-        } else {
-            return Optional.empty();
+            return Optional.
+                of(data.get().getValue(Keys.SIGN_LINES).get().get(0));
         }
+        return Optional.empty();
     }
 
     public boolean setFirstLine(TileEntity entity, Text line) {
-        if (entity.isCompatible(SignData.class)) {
+        if (entity.supports(SignData.class)) {
             SignData sign = entity.getOrCreate(SignData.class).get();
-            sign.setLine(0, line);
+            sign.set(sign.getValue(Keys.SIGN_LINES).get().set(0, line));
```

Comment:
line will not work as the lists `set()` method will definitly not return a `ListValue`.
use `sign.set(sign.lines().set(0, line));`